### ISSUE TITLE
WPF Toolkit and Futures

### ISSUE THREAD

**Issue by Thieum** [dotnet/wpf#887]
WPF Toolkit and Futures (and a few other goodies) located at https://archive.codeplex.com/?p=wpf were the first parts of wpf to be open-sourced AFAIK.

Are there plans to merge those back in this repository in any form?

**Comment by shaggygi** [comment]
I'm not sure if this is related, but there is a big focus on [WinUI here](https://github.com/Microsoft/microsoft-ui-xaml).

**Comment by Thieum** [comment]
@shaggygi I don't think it's related. WPF Toolkit was a set of extra WPF UI controls, not part of the framework, but maintained by Microsoft. There was the assumption at a time that WPF Toolkit was some sort of pre-release for new controls, that would make it some day in the framework like explained on the website:

> The WPF Toolkit is a collection of WPF features and components that are being made available outside of the normal .NET Framework ship cycle. The WPF Toolkit not only allows users to get new functionality more quickly, but allows an efficient means for giving feedback to the product team.

**Comment by shaggygi** [comment]
@Thieum Understood, but I think it will be the area they (both MS and community) will port and and new controls in the future.  There is also the Windows Community Toolkit.  Not sure how/if that will converge with WinUI or not.  Lastly, there is the Extended WPF Toolkit, but I believe is owned by another entity.

Good grief!  So much code, so little time! :smile:

**Comment by Thieum** [comment]
@shaggygi yep the [extended WPF Toolkit](https://github.com/xceedsoftware/wpftoolkit) was started by @brianlagunas then transfered to xceed. At some point, Brian made a point to not create overlaps with the original MS WPF toolkit. I don't know if this rule is still applied by xceed.

I understand your point about uwp / winui for the future. But some legacy projects are still relying on the WPF Toolkit. [One ](https://www.nuget.org/packages/WPFToolkit/) of the many unofficial nuget package has about 200 downloads a day.

**Comment by brianlagunas** [comment]
@Thieum ahh... the good'ol days of the Extended WPF Toolkit. That bring backs memories 😃. You know... I kind of miss working on that.

**Comment by onovotny** [comment]
A lot of these could also go into the Windows Toolkit. That has space for Win32/WPF controls; it's not just UWP.

I would recommend creating issues in the repo for each of the controls from the old toolkit that you'd like to see come forward: https://github.com/windows-toolkit/WindowsCommunityToolkit

**Comment by Thieum** [comment]
@onovotny Thanks for the info! I'm not fully familiar with the organisation of the Windows community toolkit. Would those controls go in https://github.com/windows-toolkit/Microsoft.Toolkit.Win32 to be used directly from wpf or in https://github.com/windows-toolkit/WindowsCommunityToolkit and used through a xaml island?

**Comment by onovotny** [comment]
Probably the Win32 one?
